data/data/bootstrap/files/usr/local/bin/bootstrap-verify-api-server-urls.sh

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+
+# shellcheck disable=SC1091
+. /usr/local/bin/bootstrap-service-record.sh
+
+function check_api_url {
+    record_service_stage_start "check-api-url"
+
+    API_URL="$(oc get Infrastructure -o json | jq -r '.items | .[] | .status | ."apiServerURL"' | sed $'s/:/\\\n/g' | sed -n 2p | sed -e's,^\(.*//\),,g')"
+    if [[ -z "${API_URL}" ]] ; then
+        echo "API URL is not available from the Infrastructure CR"
+        record_service_stage_failure
+        exit 1
+    fi
+
+    API_IP=$(dig "$API_URL" +short)
+    if [[ ! -z "${API_IP}" ]] ; then
+        echo "Successfully resolved API URL $API_URL to $API_IP"
+    else
+        echo "Unable to resolve API URL $API_URL"
+        record_service_stage_failure
+        exit 1
+    fi
+
+    API_CURL_URL="https://${API_IP}:6443/version"
+    if [[ $(curl --head -k --silent --fail --write-out "%{http_code}\\n" "${API_CURL_URL}" -o /dev/null) == 200 ]]; then
+	echo "Success while trying to reach API URL's https endpoint"
+	record_service_stage_success
+    else
+	echo "Unable to reach API URL's https endpont"
+	record_service_stage_failure
+    fi
+}
+
+function check_api_int_url {
```

Comment:
There is a lot of duplication. Can we just have one function which accepts a URL and then check whether we can resolve/curl from it?

Reply:
Went through that exercise and saw very small benefits. So, will do that if I need to come up with a diff patch for some other reason.

Reply:
Its not about convenience. My comment was about good practice. Do it right the first time rather than fix it later.